src/commands/watch/uppificator.rs

```diff
@@ -0,0 +1,110 @@
+use std::path::PathBuf;
+use uuid::Uuid;
+
+pub(crate) struct Uppificator {
+    pub spin_bin: PathBuf,
+    pub up_args: Vec<String>,
+    pub manifest: PathBuf,
+    pub clear_screen: bool,
+    pub watched_changes: tokio::sync::watch::Receiver<Uuid>,
+    pub pause_feed: tokio::sync::mpsc::Receiver<Pause>,
+    pub stopper: tokio::sync::watch::Receiver<Uuid>,
+}
+
+#[derive(Debug)]
+pub(crate) enum Pause {
+    Pause,
+    Unpause,
+}
+
+enum UppificatorAction {
+    Restart,
+    Resume,
+    Stop,
+}
+
+impl Uppificator {
+    pub(crate) async fn run(&mut self) {
+        'run: loop {
+            let mut cmd = tokio::process::Command::new(&self.spin_bin);
+            cmd.arg("up")
+                .arg("-f")
+                .arg(&self.manifest)
+                .args(&self.up_args);
+            let mut child = match cmd.spawn() {
+                Ok(ch) => ch,
+                Err(e) => {
+                    tracing::error!("Can't launch `spin up`: {e:#}");
+                    break 'run;
+                }
+            };
+
+            let mut resuming_after_build = false;
+
+            loop {
+                match self.next_event(&mut child).await {
+                    UppificatorAction::Restart => break,
+                    UppificatorAction::Resume => {
+                        resuming_after_build = true;
+                        continue;
+                    }
+                    UppificatorAction::Stop => break 'run,
+                }
+            }
+
+            if self.clear_screen && !resuming_after_build {
+                _ = clearscreen::clear();
+            }
+        }
+    }
+
+    async fn next_event(&mut self, child: &mut tokio::process::Child) -> UppificatorAction {
+        tokio::select! {
+            _ = child.wait() => {
+                UppificatorAction::Restart
+            },
+            _ = self.watched_changes.changed() => {
+                stop(child).await;
+                UppificatorAction::Restart
+            },
+            p = self.pause_feed.recv() => {
+                if matches!(p, Some(Pause::Pause)) {
+                    loop {
+                        match self.pause_feed.recv().await {
+                            Some(Pause::Unpause) => return UppificatorAction::Resume,
+                            _ => continue,
+                        }
+                    }
+                } else {
+                    UppificatorAction::Resume
+                }
+            }
+            _ = self.stopper.changed() => {
+                stop(child).await;
+                UppificatorAction::Stop
+            }
+        }
+    }
+}
+
+#[cfg(unix)]
```

Comment:
Now that we're manually doing the process management and killing what benefit are we getting from `watchexec`? Did Notify suite our needs? If we could it'd be nice to get rid of a dependency.

Reply:
`watchexec` layers filtering on top of `notify` and I couldn't see a way to use filterers with raw notify.  (My attempt to use raw notify is why that leftover dependency.)  If we can do that then I'm totally up for it.

Reply:
Let's just punt on it then. It seems like it is still providing us enough value.